### PR TITLE
IP address / Port for Validator

### PR DESCRIPTION
**Pull request by owl352**
# Issue
Right now, we only have protx hash available from Tenderdash RPC, the host and port of the node is not implemented, so this info is only available in the Core chain.

Users should be able to see ip address and ports of the services running on this validat

# Things done
`DashCoreRPC` class was created (Like `TenderdashRPC`).
`Validator` class has been modified, and the `ValidatorsController` has also been modified.
I also updated `README.md` and corrected it by adding the missing sections
Now we get this response for the validators
```
{
  proTxHash: "F60A6BF9EC0794BB0CFD1E0F2217933F4B33EDE6FE810692BC275CA18148AEF0",
  isActive: true,
  proposedBlocksAmount: 5,
  lastProposedBlockHeader: {
    height: 5,
    timestamp: "2024-06-23T13:51:44.154Z",
    hash: "7253F441FF6AEAC847F9E03672B9386E35FC8CBCFC4A7CC67557FCA10E342904",
    l1LockedHeight: 1337,
    appVersion: 1,
    blockVersion: 13,
    validator: "F60A6BF9EC0794BB0CFD1E0F2217933F4B33EDE6FE810692BC275CA18148AEF0"
  },
  proTxInfo: {
    "type": "Evo",
    "collateralHash": "6ce8545e25d4f03aba1527062d9583ae01827c65b234bd979aca5954c6ae3a59",
    "collateralIndex": 19,
    "collateralAddress": "yYK3Kiq36Xmf1ButkTUYb1iCNtJfSSM4KH",
    "operatorReward": 0,
    "confirmations": 214424,
    "state": {
        "version": 2,
        "service": "35.164.23.245:19999",
        "registeredHeight": 850334,
        "lastPaidHeight": 1064721,
        "consecutivePayments": 0,
        "PoSePenalty": 0,
        "PoSeRevivedHeight": 1027671,
        "PoSeBanHeight": -1,
        "revocationReason": 0,
        "ownerAddress": "yWrbg8HNwkogZfqKe1VW8czS9KiqdjvJtE",
        "votingAddress": "yWrbg8HNwkogZfqKe1VW8czS9KiqdjvJtE",
        "platformNodeID": "b5f25f8f70cf8d05c2d2970bdf186c994431d84e",
        "platformP2PPort": 36656,
        "platformHTTPPort": 1443,
        "payoutAddress": "yeRZBWYfeNE4yVUHV4ZLs83Ppn9aMRH57A",
        "pubKeyOperator": "b928fa4e127214ccb2b5de1660b5e371d2f3c9845077bc3900fc6aabe82ddd2e61530be3765cea15752e30fc761ab730"
    }
  }
}
```

### CODE REVIEW

File: packages/api/.env [hunk 1]
```diff
@@ -2,5 +2,8 @@ POSTGRES_HOST=127.0.0.1
 POSTGRES_DB=indexer
 POSTGRES_USER=indexer
 POSTGRES_PASS=indexer
-BASE_URL=http://127.0.0.1:36657
-EPOCH_CHANGE_TIME=3600000
+TENDERDASH_URL=http://127.0.0.1:36657
+DASHCORE_URL=http://127.0.0.1:19998
+DASHCORE_USER=8JjXoVNu
```
Comment (pshenmic) [inline packages/api/.env:7]: username

File: packages/api/.env [hunk 2]
```diff
@@ -2,5 +2,8 @@ POSTGRES_HOST=127.0.0.1
...
+DASHCORE_PASS=mCq47H4wllv7
```
Comment (pshenmic) [inline packages/api/.env:8]: password

File: packages/api/src/dashcoreRpc.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const fetch = require('node-fetch')
+const ServiceNotAvailableError = require('./errors/ServiceNotAvailableError')
+
+const call = async (method, params, id) => {
```
Comment (pshenmic) [inline packages/api/src/dashcoreRpc.js:4]: Maybe change it to https://www.npmjs.com/package/@dashevo/dashd-rpc ?

**Review by pshenmic — CHANGES_REQUESTED** [review]

File: packages/api/src/controllers/ValidatorsController.js [hunk 4]
```diff
@@ -46,13 +65,28 @@ class ValidatorsController {
       activeValidators
     )
 
+    for (let i = 0; i < validators.resultSet.length; i++) {
```
Comment (pshenmic) [inline packages/api/src/controllers/ValidatorsController.js:68]: Use map + promise all

File: packages/api/src/models/Validator.js [hunk 5]
```diff
@@ -6,17 +6,19 @@ module.exports = class Validator {
   isActive
   proposedBlocksAmount
   lastProposedBlockHeader
-
+  proTxInfo
```
Comment (pshenmic) [inline packages/api/src/models/Validator.js:9]: Add a model for proTxInfo

File: packages/api/src/dashcoreRpc.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  host: process.env.DASHCORE_HOST,
+  port: Number(process.env.DASHCORE_PORT),
+  user: process.env.DASHCORE_USER,
+  pass: process.env.DASHCORE_PASS
+}
+
+const rpc = new RpcClient(config)
+
+class DashCoreRPC {
+  static async getValidatorInfo (proTxHash) {
```
Comment (pshenmic) [inline packages/api/src/dashcoreRpc.js:15]: getProTxInfo

File: packages/api/test/integration/validators.spec.js [hunk 7]
```diff
@@ -49,6 +50,8 @@ describe('Validators routes', () => {
       async () =>
         Promise.resolve(activeValidators.map(activeValidator =>
           ({ pro_tx_hash: activeValidator.pro_tx_hash }))))
+
+    mock.method(DashCoreRPC, 'getValidatorInfo', async () => null)
```
Comment (pshenmic) [inline packages/api/test/integration/validators.spec.js:54]: Maybe mock dashd-rpc methods instead? Check if that's possible and not too hard

File: packages/api/test/integration/validators.spec.js
Comment (pshenmic) [inline packages/api/test/integration/validators.spec.js:1]: Add a test case for failed call to DashCore RPC / Tenderdash

**Review by pshenmic — CHANGES_REQUESTED** [review]

**Review by pshenmic — CHANGES_REQUESTED** [review]

File: packages/frontend/src/app/api/content.md [hunk 8]
```diff
@@ -183,6 +192,32 @@ GET /validators
         appVersion: 1,
         blockVersion: 13
         validator: "F60A6BF9EC0794BB0CFD1E0F2217933F4B33EDE6FE810692BC275CA18148AEF0"
+      },
+      proTxInfo: {
+        "type": "Evo",
```
Comment (pshenmic) [inline packages/frontend/src/app/api/content.md:197]: Keep field names without "", like in other places

**Review by pshenmic — CHANGES_REQUESTED** [review]

File: packages/frontend/src/app/api/content.md (hunk 8 above)
Comment (owl352) [reply packages/frontend/src/app/api/content.md]: Done

**Review by pshenmic — APPROVED** [review]

**Pull request merged**